java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -470,6 +474,27 @@ public static HostColumnVector fromStrings(String... values) {
     });
   }
 
+  /**
+   * Create a new vector from the given values.  This API supports inline nulls, but it is inefficient.
+   * Notice all input BigDecimals should share same scale.
+   */
+  public static HostColumnVector fromDecimals(BigDecimal... values) {
```

Comment:
I am a little concerned about this API and would like feedback on it. By default `BigDecimal` sets the scale dynamically to match best the floating point value passed in and similarly with the precision, unless you pass in a MathContex but that is only for precesion.

https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#BigDecimal-double-

This is the same for string representations too.  This means that as a user I will likely have to set the scale or every BigDecimal I create.

```
HostColumnVector.fromDecimals(new BigDecimal(1.1).setScale(1), new BigDecimal(1.2).setScale(1))
```

Along with this there is no way for me to set the precision on a BigDecimal.  There is no way for me to store all 0 values as a DECIMAL64.  I would have to cast it after creating it.  All of this feels cumbersome when what I want to do is to create a column for testing as quickly as possible.

Would it be better to have an API that takes a scale and precision(32/64 bit) followed by a list of floating point values instead/in addition to this one?

Reply:
With new method `decimalFromLongs(int scale, long... unscaledValue)`,  we can store all `0L` values into a columnVector backed by DECIMAL64. Another method `decimalFromInts(int scale, int... unscaledValue)` enables creating columnVector with DECIMAL32.
In terms of floating point values, I am not sure whether to round them in JVM or in GPU side. The former one looks inefficient, the latter one needs  the support of `round decimal`  in libcudf ?

Reply:
> Along with this there is no way for me to set the precision on a BigDecimal. There is no way for me to store all 0 values as a DECIMAL64. 

I thought about this before, but there technically is a way to build it.  You'd have to do somehting like this:
```java
ColumnVector.build(DType.create(DTypeEnum.DECIMAL64, myScale), myDecimals.length, (b) -> b.appendArray(myDecimals))
```
Not super simple, but it is possible.

However the main point of BigDecimal scale being problematic since it can adjust during computation is well taken.  Should the BigDecimal form automatically find the largest scale, using that for the scale and adjusting the scales of all other BigDecimal values to match, using the maximum precision after scale adjustments to determine the underlying cudf type?  And/Or should the user specify the cudf decimal type along with the list of BigDecimals?  Or should we just ditch BigDecimal and have the user translate them to int/long values themselves?  :smile:

Reply:
So there are two use cases here that we have to worry about.  One of them is the advanced API where what we care about is efficiency.  These are the ones where we are moving around ints or longs and making it all fit. In that case we are appending a row at a time some buffering up all the decimal values in an array and then putting them in with

```
ColumnVector.build(DType.create(DTypeEnum.DECIMAL64, myScale), myDecimals.length, (b) -> b.appendArray(myDecimals))
```

Is not going to work well.

The second set of APIs need to be about testing and making it clean and simple to build exactly what we want for testing.  The main point of my question is not CAN we do it, but is there a cleaner way to do it.  How do you want the tests to look?  The main question for me is around floating point and do we want to deal with floating point values when working with decimal or do we just want to use Strings for it?  Fun Fact when BigDecimal converts from a float it turns the float into a String first.

Reply:
I tried to add method 
```java
public static ColumnVector decimalFromDoubles(DType.DTypeEnum type, int scale, double... values);
```
to deal with floating point values. The basic idea is from @revans2 's suggestion "an API that takes a scale and precision(32/64 bit) followed by a list of floating point values".
Instead of converting floating point values to strings, `decimalFromDoubles` rescales float numbers and extracts integral part directly. Since we don't have to handle float numbers who overflows Int_64.